src/router.js

```diff
@@ -28,3 +33,28 @@ const router = new Router({
 })
 
 export default router
+
+// wait for a value to be available
+async function waitForAvailable(selectorFn) {
+  if (selectorFn() === null || selectorFn() === undefined) {
+    await new Promise(resolve =>
+      setTimeout(async () => {
+        await waitForAvailable(selectorFn)
+        resolve()
+      }, 50)
+    )
+  }
+}
+
+// check if feature is allowed and redirect if not
+async function featureAvailable(store, to) {
+  await waitForAvailable(() => store.state.networks.network)
```

Comment:
I think there is a better way to do this.

The router guards will wait on promises automatically, and Apollo is reactive I believe, and will return some kind of promise based value I think? I haven't checked it, but I think we can just plug one into the other. The `waitForAvailable` doesn't feel right to me.